src/logic.py

```diff
@@ -182,3 +187,68 @@ def dispatch_css_maintenance_action(maintenance_mode: bool):
             logger.info('CSS GH API status: %s' % x.status_code)
         else:
             logger.error('GH API error: %s' % x.content)
+
+
+def dispatch_uptime_incident(enable_incident: bool):
+    global incident_id
+
+    url = config.get('uptime_status_api')
+    uptime_token = config.get('uptime_status_token')
+    uptime_statuspage_id = config.get('uptime_status_page_id')
+    namespace = config.get('namespace')
+    env = namespace[7:]
+    if uptime_token == "" or uptime_statuspage_id == "":
+        logger.error('The uptime status page incident creation/closure has not been configured')
+        return
+
+    incident_url = f"{url}{uptime_statuspage_id}/incidents/"
+    bearer = 'token %s' % uptime_token
+    headers = {'Authorization': bearer}
+
+    body = {
+        "name": f"[{env}] Keycloak is in Disaster Recovery mode",
+        "include_in_global_metrics": False,
+        "updates": [
+            {
+                "id": 0,
+                "description": f"The switchover to GoldDR was triggered for the {env} environment.",
+                "incident_state": "investigating"
+            }
+        ],
+        "incident_type": "INCIDENT",
+        "update_component_status": True,
+        "notify_subscribers": True
+    }
+
+    if enable_incident:
+        try:
+            x = requests.post(incident_url, json=body, headers=headers)
```

Comment:
@thegentlemanphysicist if there is already an existing incident, do we still need to post one?? Does the uptime page support multiple incidents reporting at the same time?

Reply:
It does support multiple incidents, though it's a good question about wether or not we should be creating multiple ones since it cannot close multiple ones if 2 or more are open.  In normal function of the switchover agent that shouldn't happen, but I can add a check that logs an error if it happens